dbt/adapters/databricks/connections.py

```diff
@@ -449,12 +450,76 @@ def fetchone(self) -> Optional[Tuple]:
         return self._cursor.fetchone()
 
     def execute(self, sql: str, bindings: Optional[Sequence[Any]] = None) -> None:
+        # print(f"execute: {sql}")
         if sql.strip().endswith(";"):
             sql = sql.strip()[:-1]
         if bindings is not None:
             bindings = [self._fix_binding(binding) for binding in bindings]
         self._cursor.execute(sql, bindings)
 
+        # if the command was to refresh a materialized view we need to poll
+        # the pipeline until the refresh is finished.
+        refresh_search = re.search(r"refresh\s+materialized\s+view\s+([`\w.]+)", sql)
+        if refresh_search:
+            view_name = refresh_search.group(1).replace("`", "")
+            self.pollMaterializedViewPipeline(view_name)
+
+    def pollMaterializedViewPipeline(
+        self,
+        view_name: str,
+    ) -> None:
+        # interval in seconds
+        polling_interval = 10
+
+        # timeout in seconds
+        # TODO: update to longer value
+        timeout = 60 * 5
+
+        stopped_states = ("DELETED", "FAILED", "IDLE")
+        host = self._cursor.connection.host
+        headers = self._cursor.connection.thrift_backend._auth_provider._header_factory()
+
+        url = f"https://{host}/api/2.1/unity-catalog/tables/{view_name}"
+        resp1 = requests.get(url, headers=headers)
```

Comment:
we have to add auth here so to avoid connection files and env variables from request